runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/WindmillEndpoints.java

```diff
@@ -132,6 +132,10 @@ private static Optional<HostAndPort> tryParseDirectEndpointIntoIpV6Address(
             directEndpointAddress.getHostAddress(), (int) endpointProto.getPort()));
   }
 
+  public final boolean isEmpty() {
+    return equals(none());
```

Comment:
How about changing none() to return some singleton instead of building every time if we might be calling empty a lot.

Reply:
removed but made none a singleton